### PR TITLE
Added Security to the project

### PR DESCRIPTION
**Pull request by holsoni**

### CODE REVIEW

File: src/main/java/com/example/bookstore/service/UserService.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.example.bookstore.service;
+
+import com.example.bookstore.dto.user.UserRegistrationRequestDto;
+import com.example.bookstore.dto.user.UserResponseDto;
+import com.example.bookstore.exception.RegistrationException;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (satori11111) [inline src/main/java/com/example/bookstore/service/UserService.java]: useless annotation

File: src/main/resources/db/changelog/changes/05-create-users-roles-table.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+                  type: bigint
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: role_id
+                  type: bigint
+                  constraints:
+                    nullable: false
```
Comment (satori11111) [inline src/main/resources/db/changelog/changes/05-create-users-roles-table.yaml]: add foreign Key constraint to columns and i think both columns should be primary Key

File: src/main/resources/db/changelog/changes/04-create-role-table.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: name
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+                    unique: true
```
Comment (satori11111) [inline src/main/resources/db/changelog/changes/04-create-role-table.yaml]: add newLine

File: src/main/java/com/example/bookstore/exception/RegistrationException.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.example.bookstore.exception;
+
+public class RegistrationException extends RuntimeException {
```
Comment (satori11111) [inline src/main/java/com/example/bookstore/exception/RegistrationException.java]: imo need to extends Exception because you throw it in method signature

**Review by satori11111 — APPROVED** [review]
Brilliant!! But look at my comments and think about them

File: src/main/resources/db/changelog/changes/05-create-users-roles-table.yaml (hunk 2 above)
Comment (vovasalyha) [reply src/main/resources/db/changelog/changes/05-create-users-roles-table.yaml]: Yes, this 2 fields should form a composite primary key

File: src/main/java/com/example/bookstore/service/UserService.java (hunk 1 above)
Comment (vovasalyha) [reply src/main/java/com/example/bookstore/service/UserService.java]: agree

File: src/main/java/com/example/bookstore/exception/RegistrationException.java (hunk 4 above)
Comment (vovasalyha) [reply src/main/java/com/example/bookstore/exception/RegistrationException.java]: agree, make it checked exception, or do not declare in method signature at all

File: src/main/java/com/example/bookstore/config/SecurityConfig.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@RequiredArgsConstructor
+@Configuration
+public class SecurityConfig {
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/config/SecurityConfig.java:25]: Are you sure security is enabled?

**Review by vovasalyha — CHANGES_REQUESTED** [review]
Please take a look on the comments

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**